cpp/bench/ann/src/common/conf.h

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) 2023, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+#include <iostream>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#define JSON_DIAGNOSTICS 1
+#include <nlohmann/json.hpp>
+
+namespace raft::bench::ann {
+
+class Configuration {
+ public:
+  struct Index {
+    std::string name;
+    std::string algo;
+    nlohmann::json build_param;
+    std::string file;
+    std::vector<int> dev_list;
+
+    int batch_size;
+    int k;
+    int run_count;
+    std::vector<nlohmann::json> search_params;
+    std::string search_result_file;
+    float refine_ratio{0.0f};
+  };
+
+  struct DatasetConf {
+    std::string name;
+    std::string base_file;
+    // use only a subset of base_file,
```

Comment:
I recommend to add another config parameter to describe data type. This can be deduced from the file name.
```suggestion
    data type of datastet, possible values ["float", "int8", "uint8"]
    std::string dtype;
    // use only a subset of base_file,
```